clang/lib/Sema/SemaDecl.cpp

```diff
@@ -3302,12 +3306,8 @@ bool Sema::MergeFunctionDecl(FunctionDecl *New, NamedDecl *&OldD,
     }
   }
 
-  if (New->hasAttr<ReqdWorkGroupSizeAttr>() &&
-      Old->hasAttr<ReqdWorkGroupSizeAttr>())
     checkDimensionsAndSetDiagnostics<ReqdWorkGroupSizeAttr>(*this, New, Old);
```

Comment:
I understand this is what @erichkeane suggested earlier, but I am not sure I like this.  Would we do this for every attribute we want to merge?   Seems like a lot of wasted calls.

Reply:
I think it ends up being basically the same runtime-cost here after optimization.  It seems like an easy function to inline, and hasAttr and getAttr have identical runtime cost.

My fear here is that this ends up being REALLY error-prone as time goes on, we're going to be adding these for a handful of attributes and it becomes easy to copy/paste error the THREE uses of the attribute type.